### PR TITLE
Add appId and version to applications as metadata

### LINKED ISSUE #88: Change ``isDelegate`` in the Register and Deregister events to a unique app id
Right now there's no way of looking up apps in the metadata repository because the only things provided on the ``Register`` event are the computed ID and the address of the organ or app.

The issue with this is that multiple apps or organs can have the same computed ID, since it's just a hash of the function signatures in that particular contract.

The provided address can be a shared instance between multiple DAOs or a single instance for that specific DAO, so we can't use the address either.

My proposal is that we change ``isDelegate`` to something like ``appId``, where the ``appId`` is a unique identifier for the app in question (i.e. even if two apps A and B have the same ``compId``, they will have a different ``appId``).

For organs the ``appId`` would be nil.

### PR DESCRIPTION
**Pull request by izqui**
Alternative implementation to close #88. 

Instead of changing the `register` function logic when the client sees a new app being added it can just do `app.appId()` or `app.version()`.

@onbjerg opinions?

### CODE REVIEW

**Comment by onbjerg** [comment]
This is fine 🥇

File: contracts/apps/IApplication.sol [hunk 1]
```diff
@@ -4,4 +4,6 @@ import "../misc/DAOMsg.sol";
 
 contract IApplication is DAOMsgReader {
     function init() internal;
+    function appId() constant returns (string);
+    function version() constant returns (string);
```
[the remark this answers is not in the document]
Comment (onbjerg) [reply contracts/apps/IApplication.sol]: I suppose that's a bit better, makes comparing versions easier.

**Comment by onbjerg** [comment]
No longer a valid PR since the architecture changed. Close @izqui?

**Comment by izqui** [comment]
No longer relevant indeed

**Pull request closed without merging**